install/build.py

```diff
@@ -164,13 +164,26 @@ def get_compiler_setting(use_hip):
         else:
             define_macros.append(('CUPY_NO_NVTX', '1'))
 
+    # For CUB, we need the complex and CUB headers. CuPy has bundled CUB, but
+    # if the env var is set, we allow overwriting. The search precedence is:
+    #   1. User-provided CUB path, if any
+    #   2. CuPy's CUB bundle
+    #   3. built-in CUB (for CUDA 11+)
     cub_path = os.environ.get('CUB_PATH', '')
+    if cub_path:
+        utils.print_warning('CUB_PATH is DEPRECATED!',
+                            'CUB_PATH: ' + cub_path,
+                            'It will be used now, but in the future',
+                            'only CUPY_CUB_PATH will be honored')
+    else:
+        cub_path = os.environ.get('CUPY_CUB_PATH', '')
+    cupy_header = os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                               '../cupy/core/include')
     if os.path.exists(cub_path):
-        # for <cupy/complex.cuh>
-        cupy_header = os.path.join(os.path.dirname(os.path.realpath(__file__)),
-                                   '../cupy/core/include')
-        include_dirs.append(cupy_header)
-        include_dirs.append(cub_path)
+        include_dirs.insert(0, cub_path)
+    else:
+        include_dirs.insert(0, os.path.join(cupy_header, 'cupy/cub'))
+    include_dirs.insert(1, cupy_header)
```

Comment:
Hi @anaruse FYI, this is what I was referring to in your PR #3405.

Reply:
Thanks for the information. It looks good to me if CUB is enabled by default.
My patch for CUDA11 still assumes that CUB is disabled by default, but with CUDA11, CUB is automatically enabled. So, I think it's time to enable CUB by default.

Reply:
I think it depends on what you mean by enabled/disabled? In this PR, the `cupy.cuda.cub` module is built by default (as indicated in the PR title), but the flag `cupy.cuda.cub_enabled` is set to `False` for backward compatibility (see the discussion very early on in this PR). The final decision is up to @kmaehashi and I'm happy to change as needed 🙂 

By the way, leaving a note for my future self: We still need CUB bundled in CuPy because CUDA 9.x and 10.x do not come with CUB, and I assume they will still be supported in the next 2-3 yrs. Once the 10.x support is dropped, we can remove the bundle (but still keep `CUPY_CUB_PATH` for overwriting).